### DISCUSSION TITLE
issue: isValid is updated to true with clearErrors even if there is other errors

### DISCUSSION THREAD

**Discussion by val2207** [react-hook-form/react-hook-form#7469] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-clearerrors-forked-pt-2-ghwwx?file=/src/index.jsx

### Steps to reproduce

1. Open console log 
2. Clear first name errors 

Result:  IsValid is at true even if other fields has errors. 

This errors is in relation with changes here and since v7.17.5 : https://github.com/react-hook-form/react-hook-form/issues/6834

In this situation if you continue to edit fields isValid will back with the correct value when another validation is tigger. But...

Here I don't know if this is directly in relation this error but in this example here (https://codesandbox.io/s/react-hook-form-v7-clearerrors-forked-pzyt0?file=/src/index.jsx) IsValid is wrongly set at true like the first example but in this situation if I continue to edit fields the value isn't anymore updated. 

### Expected behaviour

Is valid must be at true with clearErrors just if the whole form doesn't contains errors. (RHF doc, isValid: Set to true if the form doesn't have any errors.)

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> This function will update the following formState, note this has no effect on your validation rules, it only temporarily marks your errors as an empty object and set form to valid.
> - set isValid to true
> - set errors to empty object

please refer to the doc: 

<img width="970" alt="Screen Shot 2022-01-07 at 11 53 24 pm" src="https://user-images.githubusercontent.com/10513364/148546747-b1a6e1a7-bd78-4520-9eed-385d79a38909.png">

clearError will temp clear all errors and reset isValid, if you need updated isValid state, use `trigger()`

**Reply by val2207** [reply]
Yes but this is wrong to me to update isValid to true in every situation or you have to change the description of isValid that's say : "Set to true if the form doesn't have any errors" Because this is not the reality. 

And if you clear a specific field (not all fields) your errors attribute isn't empty like you say in the documentation.

**Reply by bluebill1049** [reply]
> "Set to true if the form doesn't have any errors"  

That's a valid statement, will revert the incorrect commit and update the doc.
